### PR TITLE
Create new app for argocd-monitoring

### LINKED ISSUE #135: Add ArgoCD monitoring in Prometheus
We can use servicemonitors available from [here](https://argoproj.github.io/argo-cd/operator-manual/metrics/)

### PR DESCRIPTION
**Pull request by 4n4nd**
Add monitoring rbacs
Add servicemonitor

Closes #135

### CODE REVIEW

**Comment by 4n4nd** [comment]
Creating a new app in this repo for argocd monitoring is to keep all our monitoring rbacs in one repo

**Review by anishasthana — APPROVED** [review]
I think this looks good

File: argocd-monitoring/components/rbac/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1alpha1
+kind: Component
```
Comment (tumido) [inline argocd-monitoring/components/rbac/kustomization.yaml:2]: It should not be a component, if this is the core reason for the application to be present. Each kustomize scheme should have a base which is overlayed with optional components pulled to the mix. Can you please move this to the base?

File: odh/overlays/moc/monitoring/servicemonitors/argocd-servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: monitoring.coreos.com/v1
```
Comment (tumido) [inline odh/overlays/moc/monitoring/servicemonitors/argocd-servicemonitor.yaml:2]: I think this should be a separate PR.

**Review by tumido — CHANGES_REQUESTED** [review]

File: odh/overlays/moc/monitoring/servicemonitors/argocd-servicemonitor.yaml (hunk 2 above)
Comment (tumido) [reply odh/overlays/moc/monitoring/servicemonitors/argocd-servicemonitor.yaml:2]: Scratch that.. I didn't get this bit fully. It should stay here. Sorry. :facepalm:

File: argocd-monitoring/components/rbac/kustomization.yaml (hunk 1 above)
Comment (4n4nd) [reply argocd-monitoring/components/rbac/kustomization.yaml]: moved it to base :+1:

**Comment by tumido** [comment]
@HumairAK @4n4nd 's gonna follow up with a PR moving this into https://github.com/operate-first/apps/tree/master/cluster-scope/components :+1:

**Review by tumido — APPROVED** [review]
